synth/src/cli/telemetry.rs

```diff
@@ -40,7 +100,11 @@ where
     Fut: Future<Output = Result<T, E>>,
     E: AsRef<dyn Error + 'static>
 {
-    let client = TelemetryClient::new();
+    if is_enabled() {
+        let synth_command = serde_json::to_string(&args).unwrap();
+        // Register a custom panic hook that will send a bug report to posthog
+        panic::set_hook(Box::new(move |_| { report_panic(&synth_command, &TELEMETRY_CLIENT); }));
```

Comment:
Can we have the `report_panic` hook run after the "default" panic hook? That would still print the standard panic message which could be useful.

We could use [`panic::take_hook`](https://doc.rust-lang.org/std/panic/fn.take_hook.html) and move the old hook into the new one for that.